hibernate-core/src/main/java/org/hibernate/type/format/OsonDocumentWriter.java

```diff
@@ -0,0 +1,262 @@
+				break;
+			case SqlTypes.FLOAT:
+				generator.write( javaType.unwrap( value,Float.class,options ) );
+				break;
+			case SqlTypes.REAL:
+			case SqlTypes.DOUBLE:
+				generator.write( javaType.unwrap( value,Double.class,options ) );
+				break;
+			case SqlTypes.CHAR:
+			case SqlTypes.NCHAR:
+			case SqlTypes.VARCHAR:
+			case SqlTypes.NVARCHAR:
+				if ( value instanceof Boolean ) {
+					String c = ((Boolean) value) ? "Y" : "N";
+					generator.write( c );
+					break;
+				}
+			case SqlTypes.LONGVARCHAR:
+			case SqlTypes.LONGNVARCHAR:
+			case SqlTypes.LONG32VARCHAR:
+			case SqlTypes.LONG32NVARCHAR:
+			case SqlTypes.CLOB:
+			case SqlTypes.MATERIALIZED_CLOB:
+			case SqlTypes.NCLOB:
+			case SqlTypes.MATERIALIZED_NCLOB:
+			case SqlTypes.ENUM:
+			case SqlTypes.NAMED_ENUM:
+				// correct?
+				generator.write( javaType.unwrap( value,String.class,options ) );
+				break;
+			case SqlTypes.DATE:
+				DATE dd = new DATE(javaType.unwrap( value,java.sql.Date.class,options ));
+				OracleJsonDate jsonDate = new OracleJsonDateImpl(dd.shareBytes());
+				generator.write(jsonDate);
+				break;
+			case SqlTypes.TIME:
+			case SqlTypes.TIME_WITH_TIMEZONE:
+			case SqlTypes.TIME_UTC:
+				Time time = javaType.unwrap( value, Time.class,options );
+				generator.write( time.toString() );
+				break;
+			case SqlTypes.TIMESTAMP:
+				TIMESTAMP TS = new TIMESTAMP(javaType.unwrap( value, Timestamp.class, options ));
+				OracleJsonTimestamp writeTimeStamp = new OracleJsonTimestampImpl(TS.shareBytes());
+				generator.write(writeTimeStamp);
+				break;
+			case SqlTypes.TIMESTAMP_WITH_TIMEZONE:
+				try {
+					OffsetDateTime dateTime = javaType.unwrap( value, OffsetDateTime.class, options );
+					generator.write( dateTime );
+				}
+				catch (Exception e) {
+					Timestamp tswtz = javaType.unwrap( value, Timestamp.class, options );
+					TIMESTAMP TSWTZ = new TIMESTAMP(tswtz);
+					OracleJsonTimestamp writeTimeStampWTZ = new OracleJsonTimestampImpl(TSWTZ.shareBytes());
+					generator.write(writeTimeStampWTZ);
+				}
+				break;
+			case SqlTypes.TIMESTAMP_UTC:
+				if( value instanceof OffsetDateTime ) {
+					OffsetDateTime odt = javaType.unwrap( value, OffsetDateTime.class, options );
+					generator.write( odt );
+					break;
+				}
+				else if (value instanceof Instant ) {
+					Instant instant = javaType.unwrap( value, Instant.class, options );
+					generator.write(instant.atOffset( ZoneOffset.UTC )  );
+					break;
+				}
+				generator.write( javaType.unwrap( value,String.class,options ) );
+				break;
+			case SqlTypes.NUMERIC:
+			case SqlTypes.DECIMAL:
+				BigDecimal bd = javaType.unwrap( value, BigDecimal.class, options );
+				generator.write( bd );
+				break;
+
+			case SqlTypes.DURATION:
+				Duration duration = javaType.unwrap( value, Duration.class, options );
+				generator.write( duration );
+				break;
+			case SqlTypes.UUID:
+				generator.write( UUIDJavaType.INSTANCE.unwrap( (UUID)value, byte[].class, options ) );
```

Comment:
I would prefer if we encode the UUID as string to make the value easily accessible for users and also retain compatibility regarding queries.

```suggestion
				generator.write( javaType.unwrap( value, String.class, options ) );
```